app/src/main/java/com/asfoundation/wallet/ui/onboarding/OnboardingPresenter.kt

```diff
@@ -1,74 +1,138 @@
 package com.asfoundation.wallet.ui.onboarding
 
 import android.net.Uri
+import com.asfoundation.wallet.entity.Wallet
+import com.asfoundation.wallet.interact.SmsValidationInteract
+import com.asfoundation.wallet.wallet_validation.WalletValidationStatus
+import io.reactivex.Completable
 import io.reactivex.Observable
 import io.reactivex.Scheduler
 import io.reactivex.disposables.CompositeDisposable
 import io.reactivex.functions.BiFunction
+import io.reactivex.subjects.PublishSubject
 import java.util.concurrent.TimeUnit
 
 class OnboardingPresenter(private val disposables: CompositeDisposable,
                           private val view: OnboardingView,
                           private val onboardingInteract: OnboardingInteract,
-                          private val viewScheduler: Scheduler) {
+                          private val viewScheduler: Scheduler,
+                          private val smsValidationInteract: SmsValidationInteract,
+                          private val networkScheduler: Scheduler) {
+
+  private var walletCreated: PublishSubject<Boolean> = PublishSubject.create()
```

Comment:
Better to create it on the fragment and pass it as an argument.
Or maybe you can have a method on the fragment that as the subject as return and the presenter would listen to it, this way you don’t need to pass it to the fragment